op-node/rollup/engine/events.go

```diff
@@ -40,6 +40,55 @@ func (ev ForkchoiceUpdateEvent) String() string {
 	return "forkchoice-update"
 }
 
+// PromoteUnsafeEvent signals that the given block may now become a canonical unsafe block.
+// This is pre-forkchoice update; the change may not be reflected yet in the EL.
+// Note that the legacy pre-event-refactor code-path (processing P2P blocks) does fire this,
+// but manually, duplicate with the newer events processing code-path.
+// See EngineController.InsertUnsafePayload.
+type PromoteUnsafeEvent struct {
+	Ref eth.L2BlockRef
+}
+
+func (ev PromoteUnsafeEvent) String() string {
+	return "promote-unsafe"
+}
+
```

Comment:
I wonder if we can make a factory for this pattern so that we can reduce the boilerplate when creating events.